src/plugins/vis_type_timeseries/public/application/components/vis_editor.js

```diff
@@ -50,7 +50,7 @@ export class VisEditor extends Component {
       visFields: props.visFields,
       extractedIndexPatterns: [''],
     };
-    this.onBrush = createBrushHandler(getDataStart().query.timefilter.timefilter);
+    this.onBrush = createBrushHandler(this.props.vis);
```

Comment:
pass just applyFilter method, to make it clearer what dependencies are

Reply:
@ppisljar  I'm not a big fan of this, from my point of view, this approach complicates the maintenance of the code in the future, generating a large list of small dependencies belonging to one entity